components/google-cloud/google_cloud_pipeline_components/experimental/remote/gcp_launcher/custom_job_remote_runner.py

```diff
@@ -63,92 +63,79 @@ def create_custom_job(
   """
     client_options = {"api_endpoint": location + '-aiplatform.googleapis.com'}
     client_info = gapic_v1.client_info.ClientInfo(
-        user_agent="google-cloud-pipeline-components",
-    )
+        user_agent="google-cloud-pipeline-components",)
 
     custom_job_uri_prefix = f"https://{client_options['api_endpoint']}/v1/"
 
     # Initialize client that will be used to create and send requests.
     job_client = aiplatform.gapic.JobServiceClient(
-        client_options=client_options, client_info=client_info
-    )
+        client_options=client_options, client_info=client_info)
 
     # Instantiate GCPResources Proto
     custom_job_resources = GcpResources()
-    custom_job_resource = custom_job_resources.resources.add()
 
     # Check if the Custom job already exists
     if path.exists(gcp_resources) and os.stat(gcp_resources).st_size != 0:
         with open(gcp_resources) as f:
             serialized_gcp_resources = f.read()
-            custom_job_resource = json_format.Parse(serialized_gcp_resources,
-                                                    custom_job_resource)
-            custom_job_name = custom_job_resource.resource_uri[
+            custom_job_resources = json_format.Parse(serialized_gcp_resources,
+                                                     GcpResources())
+
+            custom_job_name = custom_job_resources.resources[0].resource_uri[
```

Comment:
nit - for robustness, verify the resources size is exactly 1, the resource uri is not empty, and optionally, parse the uri using regex instead of truncating.

Reply:
done.